centos nailgun: Update to 0.9.3 version

mamedin opened the pull request:
- bump version to 0.9.3
- use ng-nailgun client binary from Ubuntu Bionic package
- add symlink to latest SNAPSHOT.jar

Connects to https://github.com/archivematica/Issues/issues/247

mamedin committed:
centos nailgun: Update to 0.9.3 version

- bump version to 0.9.3
- use ng-nailgun client name instead of ng

scollazo reviewed (changes requested):
Adding binaries to a git repository isn't a good approach.

Can you explain the background on this, and why the ng binary can't be built from sources?

For the record, the binary is built in ubuntu using:
```cc -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,now -o ng nailgun-client/ng.c ```

https://launchpadlibrarian.net/336039279/buildlog_ubuntu-artful-i386.nailgun_0.9.1-5_BUILDING.txt.gz

mamedin commented:
I tried these CC options this morning and it doesn't work (except for -Wdata-time that is newer than CentOS glibc version).

mamedin commented:
Finally I could build the `ng-nailgun` client from sources. I added 2 patches to use the `ng-nailgun` client name instead of the default `ng`.

I doubt if the problem using the original `ng` client name was:

* Related to the https://github.com/artefactual/archivematica-fpr-admin/commit/4b00092edc1a3b61b33ac941eb526da39408de75 commit 
* I was testing copying the `ng` binary to a CentOS AM environment, not taking into account that the latest nailgun package (nailgun-0.9.1-1.el7.centos.x86_64) installs 2 nailgun client binaries:
```
[artefactual@am18centos ~]$ sudo rpm -qf /bin/ng
nailgun-0.9.1-1.el7.centos.x86_64

[artefactual@am18centos ~]$ sudo rpm -qf /usr/bin/ng
nailgun-0.9.1-1.el7.centos.x86_64
```
I'll investigate it further.

I think it is a good idea to use the `ng-nailgun` client name to be consistent with the Ubuntu's client names. So it is ready for CR again.

scollazo reviewed (approved):
This approach is way better than uploading the Bionic binary to the repo.

We should think now if it makes sense to revert the change made in https://github.com/artefactual/archivematica-fpr-admin/pull/98  , as all 1.8 deployments will have the "ng-nailgun" binary.

scollazo commented:
After an update, the `/etc/systemd/system/multi-user.target.wants/fits-nailgun.service` still shows `nailgun-server-0.9.2-SNAPSHOT.jar` , but I think that's a problem with the deployment scripts. Am I right?

scollazo reviewed (commented):
Everything seems ok at first look, but after deploying the new package, I'm still getting this error:

```
Oct 30 16:04:34 archivematica-centos7 fits-ngserver.sh[8726]: java.util.concurrent.ExecutionException: java.util.concurrent.ExecutionException: java.net.SocketE
Oct 30 16:04:34 archivematica-centos7 fits-ngserver.sh[8726]: at java.util.concurrent.FutureTask.report(FutureTask.java:122)
Oct 30 16:04:34 archivematica-centos7 fits-ngserver.sh[8726]: at java.util.concurrent.FutureTask.get(FutureTask.java:206)
Oct 30 16:04:34 archivematica-centos7 fits-ngserver.sh[8726]: at com.martiansoftware.nailgun.NGInputStream.lambda$new$1(NGInputStream.java:130)
Oct 30 16:04:34 archivematica-centos7 fits-ngserver.sh[8726]: at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
Oct 30 16:04:34 archivematica-centos7 fits-ngserver.sh[8726]: at java.util.concurrent.FutureTask.run(FutureTask.java:266)
Oct 30 16:04:34 archivematica-centos7 fits-ngserver.sh[8726]: at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
Oct 30 16:04:34 archivematica-centos7 fits-ngserver.sh[8726]: at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
Oct 30 16:04:34 archivematica-centos7 fits-ngserver.sh[8726]: at java.lang.Thread.run(Thread.java:748)
```

But the "Job Characterize and extract metadata" completes sucessfuly, should we be worried?

In the past, fits wasn't configured for restart after an error, so I think this behavior is exclusive from 1.8

scollazo commented on rpm/fits/package.spec:
```diff
@@ -1,7 +1,7 @@
 Name: %{name}
 Version: %{version}
-Release: 2%{?dist}
+Release: 3%{?dist}
```
This should be part of it's own PR

mamedin commented:
Hi, I can see the same warnings and the "INFO: NGSession shutting down" in logs for AM17. 

I have removed the fits commit and starting a new PR for it. 

It is ready for CR again.

mamedin committed:
centos nailgun: add symlink to latest SNAPSHOT.jar

The fits-nailgun service will use this new symlink instead of the jar
file that changes on every nailgun update.

mamedin commented:
I added a new commit that creates a symlink to the latest SNAPSHOT.jar
  
The `fits-nailgun` service can use this symlink instead of the jar file that changes on every nailgun update.

scollazo reviewed (approved):
LGTM!
